aws_doc_sdk_examples_tools/doc_gen.py

```diff
@@ -234,3 +235,45 @@ def stats(self):
             ),
             "snippets": len(self.snippets) + len(self.snippet_files),
         }
+
+
+class DocGenEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if is_dataclass(obj):
+            return asdict(obj)
+
+        if isinstance(obj, Path):
+            return {"__path__": str(obj)}
```

Comment:
This will print global paths, which has two problems - first, it's not reproducible or verifiable. Second (and corollary) it leaks environment information. Unfortunately, that I think means DocGenEncoder needs to be stateful, such that it can keep the root(s) and remove them.

Reply:
Can you expand a bit more on this? I understand the environment info leak, but I'm not grasping the proposed solution to make it stateful.

Reply:
You'll need to remove the `/User/whoever/devel/whatever` part of each path object, which is easier said than done as there's nothing stopping people from passing in docgens from /user/me/mystuff /user/you/yourstuff /root/coolthings. So the "State" is "the list of all roots that got merged into the final docgen that we're building out" and then you'll need to either remove or normalize that.

Reply:
Resolved by setting Path entries to `null` in the serialization and deleted DocGenDecoder.